hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -399,9 +453,65 @@ protected void writeIncomingRecords() throws IOException {
     }
   }
 
+  protected SerializableRecord createCDCRecord(HoodieCDCOperation operation, String recordKey, String partitionPath,
+                                               GenericRecord oldRecord, GenericRecord newRecord) {
+    GenericData.Record record;
+    if (cdcSupplementalLoggingMode.equals(HoodieTableConfig.CDC_SUPPLEMENTAL_LOGGING_MODE_WITH_BEFORE_AFTER)) {
+      record = HoodieCDCUtils.cdcRecord(operation.getValue(), instantTime,
+          oldRecord, addCommitMetadata(newRecord, recordKey, partitionPath));
+    } else if (cdcSupplementalLoggingMode.equals(HoodieTableConfig.CDC_SUPPLEMENTAL_LOGGING_MODE_WITH_BEFORE)) {
+      record = HoodieCDCUtils.cdcRecord(operation.getValue(), recordKey, oldRecord);
+    } else {
+      record = HoodieCDCUtils.cdcRecord(operation.getValue(), recordKey);
+    }
+    return new SerializableRecord(record);
+  }
+
+  protected GenericRecord addCommitMetadata(GenericRecord record, String recordKey, String partitionPath) {
```

Comment:
Why are we adding the metadata to CDC payload?

Reply:
if the `before` value or `after` value is persisted in CDC Block, we want to make the cdc record with the all fields.

Reply:
Meta fields carry purely semantical information related to their _persistence_ by Hudi. 
These aren't the part of the record's payload and we shouldn't be carrying them w/in CDC payload.

Reply:
that makes sense. so we don't add the meta fields in it, and should remove meta fields from the old record. has updated in the separate writing-related pr.